test/test_ripser.py

```diff
@@ -45,6 +45,18 @@ def test_non_square_dist_matrix(self):
 
         with pytest.raises(Exception):
             ripser(data, distance_matrix=True)
+    
+    def test_too_many_perm(self):
+        X = np.zeros((10, 2))
+
+        with pytest.raises(Exception):
```

Comment:
you should be able to check the text in the exception is for the exception you expect as well.